game/lobby.go

```diff
@@ -751,3 +767,9 @@ func (lobby *Lobby) JoinPlayer(playerName string) *Player {
 func (lobby *Lobby) canDraw(player *Player) bool {
 	return lobby.Drawer == player && lobby.CurrentWord != ""
 }
+
+func removeAccents(s string) string {
+	t := transform.Chain(norm.NFD, runes.Remove(runes.In(unicode.Mn)), norm.NFC)
```

Comment:
Questions:
Is the order of the chain important?
What exactly do the character sets mean. The go documentation is very vague and doesn't tell me anything.

Look at `Unicode.Mn` for example:

```
var _Mn = &RangeTable{
	R16: []Range16{
		{0x0300, 0x036f, 1},
		{0x0483, 0x0487, 1},
		{0x0591, 0x05bd, 1},
		{0x05bf, 0x05c1, 2},
		{0x05c2, 0x05c4, 2},
		{0x05c5, 0x05c7, 2},
		{0x0610, 0x061a, 1},
		{0x064b, 0x065f, 1},
		{0x0670, 0x06d6, 102},
		{0x06d7, 0x06dc, 1},
		{0x06df, 0x06e4, 1},
		{0x06e7, 0x06e8, 1},
		{0x06ea, 0x06ed, 1},
		{0x0711, 0x0730, 31},
		{0x0731, 0x074a, 1},
		{0x07a6, 0x07b0, 1},
		{0x07eb, 0x07f3, 1},
		{0x07fd, 0x0816, 25},
		{0x0817, 0x0819, 1},
		{0x081b, 0x0823, 1},
		{0x0825, 0x0827, 1},
		{0x0829, 0x082d, 1},
		{0x0859, 0x085b, 1},
		{0x08d3, 0x08e1, 1},
		{0x08e3, 0x0902, 1},
		{0x093a, 0x093c, 2},
		{0x0941, 0x0948, 1},
		{0x094d, 0x0951, 4},
		{0x0952, 0x0957, 1},
		{0x0962, 0x0963, 1},
		{0x0981, 0x09bc, 59},
		{0x09c1, 0x09c4, 1},
		{0x09cd, 0x09e2, 21},
		{0x09e3, 0x09fe, 27},
		{0x0a01, 0x0a02, 1},
		{0x0a3c, 0x0a41, 5},
		{0x0a42, 0x0a47, 5},
		{0x0a48, 0x0a4b, 3},
		{0x0a4c, 0x0a4d, 1},
		{0x0a51, 0x0a70, 31},
		{0x0a71, 0x0a75, 4},
		{0x0a81, 0x0a82, 1},
		{0x0abc, 0x0ac1, 5},
		{0x0ac2, 0x0ac5, 1},
		{0x0ac7, 0x0ac8, 1},
		{0x0acd, 0x0ae2, 21},
		{0x0ae3, 0x0afa, 23},
		{0x0afb, 0x0aff, 1},
		{0x0b01, 0x0b3c, 59},
		{0x0b3f, 0x0b41, 2},
		{0x0b42, 0x0b44, 1},
		{0x0b4d, 0x0b56, 9},
		{0x0b62, 0x0b63, 1},
		{0x0b82, 0x0bc0, 62},
		{0x0bcd, 0x0c00, 51},
		{0x0c04, 0x0c3e, 58},
		{0x0c3f, 0x0c40, 1},
		{0x0c46, 0x0c48, 1},
		{0x0c4a, 0x0c4d, 1},
		{0x0c55, 0x0c56, 1},
		{0x0c62, 0x0c63, 1},
		{0x0c81, 0x0cbc, 59},
		{0x0cbf, 0x0cc6, 7},
		{0x0ccc, 0x0ccd, 1},
		{0x0ce2, 0x0ce3, 1},
		{0x0d00, 0x0d01, 1},
		{0x0d3b, 0x0d3c, 1},
		{0x0d41, 0x0d44, 1},
		{0x0d4d, 0x0d62, 21},
		{0x0d63, 0x0dca, 103},
		{0x0dd2, 0x0dd4, 1},
		{0x0dd6, 0x0e31, 91},
		{0x0e34, 0x0e3a, 1},
		{0x0e47, 0x0e4e, 1},
		{0x0eb1, 0x0eb4, 3},
		{0x0eb5, 0x0eb9, 1},
		{0x0ebb, 0x0ebc, 1},
		{0x0ec8, 0x0ecd, 1},
		{0x0f18, 0x0f19, 1},
		{0x0f35, 0x0f39, 2},
		{0x0f71, 0x0f7e, 1},
		{0x0f80, 0x0f84, 1},
		{0x0f86, 0x0f87, 1},
		{0x0f8d, 0x0f97, 1},
		{0x0f99, 0x0fbc, 1},
		{0x0fc6, 0x102d, 103},
		{0x102e, 0x1030, 1},
		{0x1032, 0x1037, 1},
		{0x1039, 0x103a, 1},
		{0x103d, 0x103e, 1},
		{0x1058, 0x1059, 1},
		{0x105e, 0x1060, 1},
		{0x1071, 0x1074, 1},
		{0x1082, 0x1085, 3},
		{0x1086, 0x108d, 7},
		{0x109d, 0x135d, 704},
		{0x135e, 0x135f, 1},
		{0x1712, 0x1714, 1},
		{0x1732, 0x1734, 1},
		{0x1752, 0x1753, 1},
		{0x1772, 0x1773, 1},
		{0x17b4, 0x17b5, 1},
		{0x17b7, 0x17bd, 1},
		{0x17c6, 0x17c9, 3},
		{0x17ca, 0x17d3, 1},
		{0x17dd, 0x180b, 46},
		{0x180c, 0x180d, 1},
		{0x1885, 0x1886, 1},
		{0x18a9, 0x1920, 119},
		{0x1921, 0x1922, 1},
		{0x1927, 0x1928, 1},
		{0x1932, 0x1939, 7},
		{0x193a, 0x193b, 1},
		{0x1a17, 0x1a18, 1},
		{0x1a1b, 0x1a56, 59},
		{0x1a58, 0x1a5e, 1},
		{0x1a60, 0x1a62, 2},
		{0x1a65, 0x1a6c, 1},
		{0x1a73, 0x1a7c, 1},
		{0x1a7f, 0x1ab0, 49},
		{0x1ab1, 0x1abd, 1},
		{0x1b00, 0x1b03, 1},
		{0x1b34, 0x1b36, 2},
		{0x1b37, 0x1b3a, 1},
		{0x1b3c, 0x1b42, 6},
		{0x1b6b, 0x1b73, 1},
		{0x1b80, 0x1b81, 1},
		{0x1ba2, 0x1ba5, 1},
		{0x1ba8, 0x1ba9, 1},
		{0x1bab, 0x1bad, 1},
		{0x1be6, 0x1be8, 2},
		{0x1be9, 0x1bed, 4},
		{0x1bef, 0x1bf1, 1},
		{0x1c2c, 0x1c33, 1},
		{0x1c36, 0x1c37, 1},
		{0x1cd0, 0x1cd2, 1},
		{0x1cd4, 0x1ce0, 1},
		{0x1ce2, 0x1ce8, 1},
		{0x1ced, 0x1cf4, 7},
		{0x1cf8, 0x1cf9, 1},
		{0x1dc0, 0x1df9, 1},
		{0x1dfb, 0x1dff, 1},
		{0x20d0, 0x20dc, 1},
		{0x20e1, 0x20e5, 4},
		{0x20e6, 0x20f0, 1},
		{0x2cef, 0x2cf1, 1},
		{0x2d7f, 0x2de0, 97},
		{0x2de1, 0x2dff, 1},
		{0x302a, 0x302d, 1},
		{0x3099, 0x309a, 1},
		{0xa66f, 0xa674, 5},
		{0xa675, 0xa67d, 1},
		{0xa69e, 0xa69f, 1},
		{0xa6f0, 0xa6f1, 1},
		{0xa802, 0xa806, 4},
		{0xa80b, 0xa825, 26},
		{0xa826, 0xa8c4, 158},
		{0xa8c5, 0xa8e0, 27},
		{0xa8e1, 0xa8f1, 1},
		{0xa8ff, 0xa926, 39},
		{0xa927, 0xa92d, 1},
		{0xa947, 0xa951, 1},
		{0xa980, 0xa982, 1},
		{0xa9b3, 0xa9b6, 3},
		{0xa9b7, 0xa9b9, 1},
		{0xa9bc, 0xa9e5, 41},
		{0xaa29, 0xaa2e, 1},
		{0xaa31, 0xaa32, 1},
		{0xaa35, 0xaa36, 1},
		{0xaa43, 0xaa4c, 9},
		{0xaa7c, 0xaab0, 52},
		{0xaab2, 0xaab4, 1},
		{0xaab7, 0xaab8, 1},
		{0xaabe, 0xaabf, 1},
		{0xaac1, 0xaaec, 43},
		{0xaaed, 0xaaf6, 9},
		{0xabe5, 0xabe8, 3},
		{0xabed, 0xfb1e, 20273},
		{0xfe00, 0xfe0f, 1},
		{0xfe20, 0xfe2f, 1},
	},
```

I have no clue what these characters are :D Therefore, it'd be quite nice to have some examples as comments in the place where it's used. Same goes for `norm.NFD` and `norm.NFC`. A rule of thumb to go by is: `Never used code that you don't understand, it's gonna bite you in the ass later on.` In this case, it's gonna bite me in the ass ^^

I found a library that does some basic sanitization: https://github.com/kennygrant/sanitize

This is much more understandable and probably works for our usecase.

Reply:
Yep, everything is not 100% clear to me (particularly about `transform.chain` args order and its functioning), but what I understand come from unicode.org documenting normalization forms (NFx):
https://unicode.org/reports/tr15/

And MN (marks, non spacing) set:
https://www.fileformat.info/info/unicode/category/Mn/list.htm

What I get is:
- NFD decompose chars like 'é' in two chars 'e' and ' ́ ' 
- Remove(unicode.Mn) removes the acute char (being in Mn set)
- and NFC recompose decomposed characters in one (putting back in place those that were not in MN set)

I will give a look to that library!